File: xPLHalScripthelper/trunk/xPLHalScriptHelper/IHalObjects.cs
using System;
using System.Collections.Generic;

namespace xPLHalScriptHelper {
    interface IHalObjects {

        bool AddRecurringEvent(Sys.RecurringEventInfo eventdata);

        bool AddSingleEvent(Sys.SingleEventInfo eventdata);

        object ClearErrorLog();

        object DeleteDeviceConfig(string vdi);

        object DeleteEvent(string tag);

        object DeleteGlobal(string globalname);

        object DeleteRule(string ruleguid);

        object DeleteScript(string scriptname);

        object GetDeviceConfig(string vdi);

        object GetDeviceConfigValue(string vdi, string configitem);

        object GetErrorLog();

        object GetEvent(string tag);

        object GetGlobal(string globalname);

        //object GetReplicationInfo();

        object GetRule(string ruleguid);

        object GetScript(string scriptname);

        object GetSetting(string setting);

        object ListDevices(string options);

        object ListEvents();

        System.Collections.Generic.Dictionary<string, string> ListGlobals();

        //object ListOptions(string setting);

        object ListRuleGroups();

        object ListRules();

        object ListRules(string groupname);

        object ListScripts();

        object ListScripts(string path);

        //object ListSettings();

        object ListSingleEvents();

        object ListSubs();

        //object ListSubs(string path);

        object PutScript(string scriptname, string script);

        object RunRule(string ruleguid);

        object RunSub(string scriptname, string parameters);

        object SendXplMessage(string msgtype, string schema, string body);

        object SendXplMessage(string msgtype, string target, string schema, string body);

        object SetGlobal(string key, string value);

        object SetRule(string ruleguid, string xml);

        //object SetSetting(string settingname, string settingvalue);
    }
}

File: xPLHal2/trunk/ScriptEngine/IHalObjects.vb
Imports System

Interface IHalObjects
    Function AddRecurringEvent(ByVal eventdata As HalObjects.RecurringEventInfo) As Boolean
    Function AddSingleEvent(ByVal eventdata As HalObjects.SingleEventInfo) As Boolean
    Function ClearErrorLog() As Object
    Function DeleteDeviceConfig(ByVal vdi As String) As Object
    Function DeleteEvent(ByVal tag As String) As Object
    Function DeleteGlobal(ByVal globalname As String) As Object
    Function DeleteRule(ByVal ruleguid As String) As Object
    Function DeleteScript(ByVal scriptname As String) As Object
    Function GetDeviceConfig(ByVal vdi As String) As Object
    Function GetDeviceConfigValue(ByVal vdi As String, ByVal configitem As String) As Object
    Function GetErrorLog() As Object
    Function GetEvent(ByVal tag As String) As Object
    Function GetGlobal(ByVal globalname As String) As Object
    Function GetReplicationInfo() As Object
    Function GetRule(ByVal ruleguid As String) As Object
    Function GetScript(ByVal scriptname As String) As Object
    Function GetSetting(ByVal setting As String) As Object
    Function ListDevices(ByVal options As String) As Object

    Function ListEvents() As Object
    Function ListGlobals() As System.Collections.Generic.Dictionary(Of String, String)
    Function ListOptions(ByVal setting As String) As Object
    Function ListRuleGroups() As Object

    Function ListRules() As Object
    Function ListRules(ByVal groupname As String) As Object

    Function ListScripts() As Object
    Function ListScripts(ByVal path As String) As Object

    Function ListSettings() As Object
    Function ListSingleEvents() As Object

    Function ListSubs() As Object
    Function ListSubs(ByVal path As String) As Object

    Function PutScript(ByVal scriptname As String, ByVal script As String) As Object
    'Function ReloadScripts() As Boolean
    Function RunRule(ByVal ruleguid As String) As Object

    Function RunSub(ByVal scriptname As String, ByVal parameters As String) As Object

    Function SendXplMsg(ByVal t As String, ByVal target As String, ByVal body As String) As Object
    Function SendXplMsg(ByVal t As String, ByVal target As String, ByVal schema As String, ByVal body As String) As Object
    Function SetGlobal(ByVal key As String, ByVal value As String) As Object
    Function SetRule(ByVal ruleguid As String, ByVal xml As String) As Object
    Function SetSetting(ByVal settingname As String, ByVal settingvalue As String) As Object
End Interface
